cmd/restic/cmd_check.go

```diff
@@ -282,15 +292,19 @@ func runCheck(ctx context.Context, opts CheckOptions, gopts GlobalOptions, args
 		Verbosef("%d additional files were found in the repo, which likely contain duplicate data.\nThis is non-critical, you can run `restic prune` to correct this.\n", orphanedPacks)
 	}
 
-	Verbosef("check snapshots, trees and blobs\n")
+	if !gopts.JSON {
+		Verbosef("check snapshots, trees and blobs\n")
+	}
 	errChan = make(chan error)
 	var wg sync.WaitGroup
 
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		bar := newProgressMax(!gopts.Quiet, 0, "snapshots")
-		defer bar.Done()
+		if !gopts.JSON {
+			bar := newProgressMax(!gopts.Quiet, 0, "snapshots")
```

Comment:
With this change, `chkr.Structure` uses the wrong instance of the `bar` variable.